modules/assignment/src/main/java/com/intuit/wasabi/assignment/impl/AssignmentsImpl.java

```diff
@@ -619,12 +616,26 @@ protected Assignment getAssignment(User.ID userID, Application.Name applicationN
                     Assignment.Status.EXPERIMENT_NOT_STARTED);
         }
 
+        Assignment assignment = getAssignment(experimentID, userID, context, userAssignments, bucketList);
+
+        // Case: Experiment expires -> existing Assignments are returned and new Assignments
+        // are returned as null assignments
         if (currentTime > experiment.getEndTime().getTime()) {
```

Comment:
@AndreaSuckro Thx for trying to keep the existing behavior for Expired->PAUSED experiment BUT this change doesn't address the case where there was existing_assignment but bucket had been emptied... 

I would suggest that in this expire-check IF condition only handle the experiment STATE update logic and leave rest of the code flow as is...